config/dynamic.c

```diff
@@ -459,6 +656,8 @@ dyn_cfg_proc(void *arg)
 		 */
 		RTE_VERIFY(FD_ISSET(dy_conf->sock_fd, &fds));
 		handle_client(dy_conf->sock_fd, dy_conf);
+
+		process_cmds_from_mailbox(dy_conf);
```

Comment:
Consider the following condition: there are a number of messages before `handle_client()` is called, so the mailbox may not be able to receive back all messages that `l_update_gt_lua_states_incrementally()` sends. This can happen when `num_succ_sent_inst > dy_conf->mailbox_burst_size` and `process_return_message()` doesn't clear enough messages.

The solution comes with two steps:
1. move `process_cmds_from_mailbox()` to the beginning of the `while` loop, so it's called even when there is no connection. This should keep the mailbox empty.
2. rename `process_cmds_from_mailbox()` to `clear_mailbox()` and add a loop in the function to only return once the mailbox is empty. This will guarantee that a new connection will always have an empty mailbox.